test/module/irohad/multi_sig_transactions/state_test.cpp

```diff
@@ -25,228 +13,338 @@ using namespace std;
-  state1 += makeTx(1, time, keypair, quorum);
-  state1 += makeTx(2, time, makeKey(), quorum);
-  ASSERT_EQ(2, state1.getTransactions().size());
+  state1 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_1", "1_1"));
+  state1 += addSignatures(
+      makeTestBatch(txBuilder(2, time)), 0, makeSignature("2", "2"));
+  state1 += addSignatures(
+      makeTestBatch(txBuilder(2, time)), 0, makeSignature("3", "3"));
+  ASSERT_EQ(2, state1.getBatches().size());
 
   auto state2 = MstState::empty();
-  state2 += makeTx(1, time, keypair, quorum);
-  state2 += makeTx(1, time, makeKey(), quorum);
-  ASSERT_EQ(1, state2.getTransactions().size());
+  state2 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_2", "1_2"));
+  state2 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_3", "1_3"));
+  ASSERT_EQ(1, state2.getBatches().size());
 
   auto completed_state = state1 += state2;
-  ASSERT_EQ(1, completed_state.getTransactions().size());
-  ASSERT_EQ(1, state1.getTransactions().size());
+  ASSERT_EQ(1, completed_state.getBatches().size());
+  ASSERT_EQ(1, state1.getBatches().size());
 }
 
+/**
+ * Tests expired completer, which checks that all transactions in batch are not
+ * expired
+ */
 class TimeTestCompleter : public iroha::DefaultCompleter {
-  bool operator()(const DataType &tx, const TimeType &time) const override {
-    return tx->createdTime() < time;
+  bool operator()(const DataType &batch, const TimeType &time) const override {
+    return std::all_of(
+        batch->transactions().begin(),
+        batch->transactions().end(),
+        [&time](const auto &tx) { return tx->createdTime() < time; });
   }
 };
 
+/**
+ * @given state with one transaction
+ * @when  call erase by time, where batch will be already expired
+ * @then  checks that expired state contains batch and initial doesn't conitain
+ */
 TEST(StateTest, TimeIndexInsertionByTx) {
-  log_->info("Insert one transaction with many signatures => erase tx by time");
-
-  auto quorum = 3u;
+  auto quorum = 2u;
   auto time = iroha::time::now();
 
   auto state = MstState::empty(std::make_shared<TimeTestCompleter>());
 
-  state += makeTx(1, time, makeKey(), quorum);
-  state += makeTx(1, time, makeKey(), quorum);
-
-  ASSERT_EQ(1, state.getTransactions().size());
+  state += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                         0,
+                         makeSignature("1_1", "1_1"));
 
   auto expired_state = state.eraseByTime(time + 1);
-  ASSERT_EQ(1, expired_state.getTransactions().size());
-  ASSERT_EQ(0, state.getTransactions().size());
+  ASSERT_EQ(1, expired_state.getBatches().size());
+  ASSERT_EQ(0, state.getBatches().size());
 }
 
+/**
+ * @given init two states
```

Comment:
-> given two states with expired transactions
when the states are merged
then the resulting state does not contain any transactions

Reply:
> given two states with expired transactions
Batches are not expired at this time.